connectors/agent/config.py

```diff
@@ -64,28 +71,78 @@ def try_update(self, source):
         the method returns False.
         """
 
+        source = unit.config.source
+
         # TODO: find a good link to what this object is.
         has_hosts = source.fields.get("hosts")
         has_api_key = source.fields.get("api_key")
         has_basic_auth = source.fields.get("username") and source.fields.get("password")
+
+        assumed_configuration = {}
```

Comment:
Could not get a better name for "temporary config storage that collects config changes that we'll try to apply"

Reply:
`tmp_cfg` 😝 
I think "assumed" works great here